lib/puppet/provider/package/pip.rb

```diff
@@ -131,4 +146,33 @@ def lazy_pip(*args)
   def install_options
     join_options(@resource[:install_options])
   end
+
+  def latest_with_new_pip
+    # Less resource intensive approach for pip version 1.5.4 and above
+    execpipe ["#{self.class.pip_cmd}", "install", "#{@resource[:name]}==versionplease"] do |process|
```

Comment:
Any reason you're using install, instead of something like
```
$ pip search sslyze
SSLyze (0.13.3)  - Fast and full-featured SSL scanner.
  INSTALLED: 0.13.5
  LATEST:    0.13.3
nassl (0.13.3)   - Experimental OpenSSL wrapper for Python 2.7 and SSLyze.
  INSTALLED: 0.13.5
  LATEST:    0.13.3
```